eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventStorageEngine.java

```diff
@@ -85,6 +85,11 @@ CompletableFuture<AppendTransaction> appendEvents(@Nonnull AppendCondition condi
      * Creates a <b>finite</b> {@link MessageStream} of {@link EventMessage events} matching the given
      * {@code condition}.
      * <p>
+     * The final entry of the stream <b>always</b> contains a {@link ConsistencyMarker} in the
+     * {@link MessageStream.Entry} it's resources, paired with a {@link org.axonframework.eventhandling.NoEventMessage}.
+     * This {@code ConsistencyMarker} should be used to construct the {@link AppendCondition} when
```

Comment:
Reading this, I am wondering if we should have a `NoEventMessage`, or alternatively: 

1. We can either make a `ConsistencyTokenTerminalEventMessage`
2. What is we are able to complete a stream with a value. So we can do `Stream.complete(ConsistencyMarker)`.

Just some thoughts. It's fine for now, but it's very implicit to carry this in the context

Reply:
I'd wager that the `NoEventMessage` is my biggest doubt about this PR too. If I follow you correctly, the introduction of `ConsistencyTokenTerminalEventMessage` also means you'd remove the marker from the resources, right?

By the way, `Stream.complete(ConsistencyMarker)` is essentially what I am doing with the `whenComplete().concatWith({single-marker-entry-stream})` flow. I am wondering how much we'd need that outside of this use case. Then again, adding it shouldn't be tough, given that I have already created the structure for it.